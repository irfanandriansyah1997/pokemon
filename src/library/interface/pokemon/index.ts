export * from './pokemon-evolution.interface';
export * from './pokemon-item.interface';
export * from './pokemon-move.interface';
export * from './pokemon-species.interface';
export * from './pokemon-util.interface';
