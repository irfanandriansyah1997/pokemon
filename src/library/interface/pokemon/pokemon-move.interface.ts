import {
  IPokemonBase,
  IPokemonFlavorTextEntries
} from './pokemon-util.interface';

/**
 * Pokemon Move Interface
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.07.30
 */
export interface IPokemonMove {
  accuracy: number;
  damage_class: Partial<IPokemonBase>;
  effect_chance: number;
  effect_entries: { effect: string }[];
  flavor_text_entries: IPokemonFlavorTextEntries[];
  id: number;
  power: number;
  pp: number;
  target: Partial<IPokemonBase>;
  type: Partial<IPokemonBase>;
}
