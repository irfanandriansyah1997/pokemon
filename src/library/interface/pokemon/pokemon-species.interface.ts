import {
  IPokemonBase,
  IPokemonFlavorTextEntries,
  IPokemonGeneral
} from './pokemon-util.interface';

/**
 * Pokemon Species Interface
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.07.30
 */
export interface IPokemonSpecies {
  base_happiness: number;
  capture_rate: number;
  color: Partial<IPokemonBase>;
  egg_groups: Partial<IPokemonBase>[];
  evolution_chain: Partial<IPokemonBase>;
  flavor_text_entries: IPokemonFlavorTextEntries[];
  genera: IPokemonGeneral[];
  growth_rate: Partial<IPokemonBase>;
  habitat: Partial<IPokemonBase>;
  hatch_counter: number;
  id: number;
  is_baby: boolean;
  is_legendary: boolean;
  is_mythical: boolean;
  shape: Partial<IPokemonBase>;
}
