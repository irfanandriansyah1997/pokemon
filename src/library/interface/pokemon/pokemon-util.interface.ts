/**
 * Pokemon Base Interface
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.07.30
 */
export interface IPokemonBase {
  id: string;
  name: string;
  url: string;
}

/**
 * Pokemon Type Interface
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.07.30
 */
export interface IPokemonType {
  slot: number;
  type: Partial<IPokemonBase>;
}

/**
 * Pokemon Type Interface
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.07.30
 */
export interface IPokemonAbilities {
  ability: Partial<IPokemonBase>;
  is_hidden: boolean;
}

/**
 * Pokemon Stat Interface
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.07.30
 */
export interface IPokemonStat {
  base_stat: number;
  effort: number;
  stat: {
    name: string;
  };
}

/**
 * Pokemon Sprites Interface
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.07.30
 */
export interface IPokemonSprites {
  back_default: string;
  front_default: string;
}

export interface IPokemonGeneral {
  genus: string;
  language: Partial<IPokemonBase>;
}

/**
 * Pokemon Flavor Text Entries
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.07.30
 */
export interface IPokemonFlavorTextEntries {
  flavor_text: string;
  language: Partial<IPokemonBase>;
  version?: Partial<IPokemonBase>;
  version_group?: Partial<IPokemonBase>;
}
