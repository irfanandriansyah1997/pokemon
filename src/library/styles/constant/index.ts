export * from './color.constant';
export * from './font-variable.constant';
