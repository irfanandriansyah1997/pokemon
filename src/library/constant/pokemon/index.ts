import { ITabActionItem } from '@/library/component/tab-action/interface';
import { IPokemonTab } from '@/library/features/pokemon-detail/interface';

export const MAX_POKEMON_STATS = 180;

export const POKEMON_TAB_ITEM: ITabActionItem[] = [
  {
    id: IPokemonTab.about,
    text: `About`
  },
  {
    id: IPokemonTab.stats,
    text: `Stats`
  },
  {
    id: IPokemonTab.evolution,
    text: `Evolution`
  }
];
