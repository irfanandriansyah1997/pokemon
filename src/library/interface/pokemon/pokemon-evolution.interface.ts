import { IPokemonBase } from './pokemon-util.interface';

/**
 * Pokemon Evolution Interface
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.07.30
 */
export interface IPokemonEvo {
  evolution_details: IPokemonEvoDetail[];
  evolves_to: IPokemonEvo[];
  species: Partial<IPokemonBase>;
}

/**
 * Pokemon Evolution Detail Interface
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.07.30
 */
export interface IPokemonEvoDetail {
  held_item: Partial<IPokemonBase>;
  item: Partial<IPokemonBase>;
  known_move: Partial<IPokemonBase>;
  known_move_type: Partial<IPokemonBase>;
  location: Partial<IPokemonBase>;
  min_affection: number;
  min_beauty: number;
  min_happiness: number;
  min_level: number;
  time_of_day: string;
  trigger: Partial<IPokemonBase>;
}
