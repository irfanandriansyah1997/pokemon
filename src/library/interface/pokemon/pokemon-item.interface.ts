import { IPokemonSpecies } from './pokemon-species.interface';
import {
  IPokemonAbilities,
  IPokemonBase,
  IPokemonSprites,
  IPokemonStat,
  IPokemonType
} from './pokemon-util.interface';

/**
 * Pokemon Interface
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.07.30
 */
export interface IPokemon extends Partial<IPokemonOptional> {
  id: number;
  image: string;
  name: string;
}

/**
 * Pokemon Optional Data
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.07.30
 */
export interface IPokemonOptional {
  abilities: IPokemonAbilities[];
  base_experience: number;
  color: string;
  height: number;
  moves: { move: Partial<IPokemonBase> }[];
  nickname: string;
  pokeSpecies: IPokemonSpecies;
  species: IPokemonBase;
  sprites: IPokemonSprites;
  stats: IPokemonStat[];
  types: IPokemonType[];
  weight: number;
}
