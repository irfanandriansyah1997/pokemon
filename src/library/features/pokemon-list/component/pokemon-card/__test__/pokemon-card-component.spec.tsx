import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { PokemonCardContainer } from '@/library/styles/pokemon.styles';

import PokemonCard from '..';

describe(`Testing Pokemon Card`, () => {
  describe(`Testing Snapshot Component`, () => {
    it(`Testing Card From Pokemon List`, () => {
      const onClickSpy = jest.fn((param) => param);
      const ui = shallow(
        <PokemonCard
          id={1}
          image="image"
          name="Bulbasaur"
          onClick={onClickSpy}
        />
      );

      expect(toJson(ui)).toMatchSnapshot();
    });

    it(`Testing Card From My Pokemon List`, () => {
      const onClickSpy = jest.fn((param) => param);
      const ui = shallow(
        <PokemonCard
          id={1}
          customName="My Bulbasaur"
          image="image"
          name="Bulbasaur"
          onClick={onClickSpy}
        />
      );

      expect(toJson(ui)).toMatchSnapshot();
    });
  });

  it(`Testing Simulate Click Card`, () => {
    const onClickSpy = jest.fn((param) => param);
    const ui = mount(
      <PokemonCard
        id={1}
        customName="My Bulbasaur"
        image="image"
        name="Bulbasaur"
        onClick={onClickSpy}
      />
    );

    ui.find(PokemonCardContainer).at(0).simulate(`click`);
    expect(onClickSpy).toHaveBeenCalledTimes(1);
    expect(onClickSpy.mock.results[0].value).toBe(`Bulbasaur`);
  });
});
