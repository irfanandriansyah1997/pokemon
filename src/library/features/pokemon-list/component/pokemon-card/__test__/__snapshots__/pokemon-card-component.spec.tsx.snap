// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Pokemon Card Testing Snapshot Component Testing Card From My Pokemon List 1`] = `
<Styled(div)
  color="#ddd"
  onClick={[Function]}
  role="button"
  tabIndex={0}
>
  <div>
    <img
      alt="Pokemon Bulbasaur"
      loading="lazy"
      src="image"
    />
    <div
      className="pokemon-text"
    >
      <Styled(p)
        color="white"
        fontSize="large"
        fontWeight={700}
      >
        #001
      </Styled(p)>
      <Styled(p)
        color="white"
        fontSize="normal"
        fontWeight={400}
      >
        My Bulbasaur
      </Styled(p)>
    </div>
    <PokeBall />
  </div>
</Styled(div)>
`;

exports[`Testing Pokemon Card Testing Snapshot Component Testing Card From Pokemon List 1`] = `
<Styled(div)
  color="#ddd"
  onClick={[Function]}
  role="button"
  tabIndex={0}
>
  <div>
    <img
      alt="Pokemon Bulbasaur"
      loading="lazy"
      src="image"
    />
    <div
      className="pokemon-text"
    >
      <Styled(p)
        color="white"
        fontSize="large"
        fontWeight={700}
      >
        #001
      </Styled(p)>
      <Styled(p)
        color="white"
        fontSize="normal"
        fontWeight={400}
      >
        Bulbasaur
      </Styled(p)>
    </div>
    <PokeBall />
  </div>
</Styled(div)>
`;
